### ISSUE TITLE
Train Loss Data Available?

### ISSUE THREAD

**Issue by drothermel** [allenai/DataDecide#4]
Thank you for open-sourcing in such detail! I see the eval results on huggingface, are the train loss curves from the training runs available?

And, is it safe to assume a pretraining procedure similar to just "Stage 1" of the Olmo 2 paper, specifically:

>  Departing from previous OLMo versions, OLMo 2 models are randomly initialized from a truncated normal distribution with a mean of 0 and a standard deviation of 0.02 (see section 3.2). After that, we run a learning rate schedule that warms up the learning rate from 0 to the peak learning rate (a hyperparameter) over 2000 steps, followed by a cosine decay calibrated to reach 10% of the peak learning rate [at the end of training].

**Comment by IanMagnusson** [comment]
(Edit) oops I just saw you were asking for _train_ loss. We haven't released that yet but I can look into doing so soon! Until then I hope the Val loss might be useful. 


Thanks so much for reaching out! We're actually just in the process of releasing the loss results. They are here (https://huggingface.co/datasets/allenai/DataDecide-ppl-results) but I haven't finished the documentation yet. The loss sets we use are from an internal version of Paloma that runs by default with our pretraining code and some of the sets (ICE) require contacting the original authors at a different organization to get access to. But I'm happy to give as much detail as I can about the loss sets to help your research. (I'm at ICML this week so might be a bit slow to respond until next week). Also note that "task loss" ie loss on downstream task answers is available in our main eval results dataset and is called correct prob.

You can find the pretraining details in this readme (https://github.com/allenai/DataDecide/tree/main/pretraining). We use a branch of the original OLMo trainer and use the model ladder implementation within it. This is in between OLMo 1 and 2 in terms of hyperparams and architecture. The hyperparams for our models are in the appendix of our paper. Happy to answer any specific questions you have about the training process.

**Comment by drothermel** [comment]
Amazing! My main question is around learning rate schedule, specifically: 
- how long warmup was for each of the models and what is the initial lr
- I assume you warmup to max lr?
- Then is it cosine annealing and to what final value of learning rate?

I see that you added optim.pt files for step-0 so I think I can extract the initial learning rate but I'm not sure how to get other info.

Thanks!

**Comment by IanMagnusson** [comment]
Here's the peak LR
<img width="1346" height="801" alt="Image" src="https://github.com/user-attachments/assets/e25c6225-8fea-47b1-8872-49b17c79646f" />

Warm up starts at 0.0 and the warmup duration and schedule are defined heuristically based on model size [in the olmo ladder script](https://github.com/allenai/OLMo/blob/7094aab0448096f4994cae881edbd629d6c2f3de/scripts/ladder.py#L241-L246) @AkshitaB and @dirkgr designed the ladder in separate work and might be able to offer more insight.
